### DISCUSSION TITLE
INS - Can a mesh be too fine?

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#20350] [Q&A Modules: Navier-Stokes] [answered]
I'm having trouble converging a steady-state INS problem when I attempt to increase the mesh refinement. I plan to use the output of the simulation as a "background flow" for another simulation, which requires a finer mesh due to the physics than the INS needs. In the txt file attached, I included both the .geo file (using gmsh to make the mesh) and the input file. The mesh refinement shown on the .geo file of 2e-4 at each point works great, but decreasing this value to even 1e-4 for some points ruins the convergence.

If anybody has any clues/suggestions on how to fix this, I would greatly appreciate it.
[ins_mesh_issue.txt](https://github.com/idaholab/moose/files/8102472/ins_mesh_issue.txt)

**Comment by GiudGiud** [comment]
Hello

You could be running into the CFL conditions limit. 
What time integration scheme are you using?

Guillaume

**Reply by ghost** [reply]
I am running with a steady executioner, not a transient. Does this question still apply?

**Reply by GiudGiud** [reply]
no it does not (time integrator is effectively implicit euler, with an infinite time step)
It's usually easier to get convergence for a flow problem with a relaxation-to-steady-state transient than with a steady solve. You could try that.
@lindsayad

**Reply by ghost** [reply]
Good to know, I will try that! Thank you

**Comment by lindsayad** [comment, marked as answer]
Have you estimated what your cell Reynolds number should be for this problem? The answer to this question will determine whether SUPG is needed or not

**Reply by ghost** [reply]
In the thinnest part of the domain, I'm getting a Reynold's number of about 50 with a Q= 1000 sccm, rho = 0.2135 kg/m^3, mu = 2.3e-5 Pa-s, and D = 0.004 m. I'm not sure if this what you mean by "cell", though.

**Reply by lindsayad** [reply]
I'm going to add a reynolds number aux kernel to the NS module to help users compute this. Should have a PR up sometime today.

For determining whether to stabilize or not we are typically interested in the highest element/cell Reynolds number, so that will occur for the largest `_current_elem->hmax()`. Anyway any Re greater than ~1 strictly speaking raises a need for stabilization technique like upwinding. You can get this for finite element INS by putting `supg = true` in your `GlobalParams` block. This may or may not make your convergence issues go away. I'm going to run your problem at some point today.

**Reply by lindsayad** [reply]
You may also consider using PSPG which negates the need to have different orders for your velocity and pressure variables. To get PSPG, put `pspg = true` in your `GlobalParams` block

**Reply by lindsayad** [reply]
See https://github.com/idaholab/moose/pull/20361 for a reynolds number aux kernel. I found that if I set `pspg = true` and used first order for all variables, I could get your simulation to converge with 1e-4 for your characteristic mesh size. However, if I took it lower to 5e-5, then even with `pspg = true` the nonlinear solve stagnated. So I would posit that the answer to your question is yes, although I'm hesitant to give a sure answer as to why. My gut feeling is that you run out of precision and the problem loses conditioning but I don't have definitive proof of this. Using quad precision could test my hypothesis

**Reply by ghost** [reply]
good to know! Did you also use the "supg = true"?

**Reply by lindsayad** [reply]
I believe I ran with both `supg = true` and `supg = false` and the nonlinear solve behavior (and results) were pretty similar in this case

**Reply by lindsayad** [reply]
So having talked with another colleague, I think a better explanation is more along the lines of @GiudGiud 's initial suggestion. In order for Newton-Raphson to be effective you must start with an initial guess that is reasonably close to the converged solution, e.g. you need to be in the "ball of attraction" or "sphere of convergence" for Newton. When you are doing a `Steady` solve with arbitrary initial conditions you are very far from the final solution. This is likely enhanced the finer and finer you make the mesh. So I suspect that if you follow a `Transient` pseudo-stepping strategy like @GiudGiud suggested, then you should be able to reach a steady-state for a finer mesh than you are able to with `Steady` with an arbitrary initial guess/conditions.
